src/pages/VenuePage/VenuePage.tsx

```diff
@@ -93,57 +94,49 @@ const VenuePage: React.FC = () => {
 
   // NOTE: User location updates
 
-  const updateUserLocationToCurrentVenue = useCallback(() => {
-    if (!userId || !venueName) return;
-
-    trackLocationEntered({ userId, locationName: venueName });
-  }, [userId, venueName]);
-
   useInterval(() => {
-    updateUserLocationToCurrentVenue();
+    if (!userId || !profile?.lastSeenIn) return;
+
+    updateCurrentLocationData({
+      userId,
+      profileLocationData: profile.lastSeenIn,
+    });
   }, LOC_UPDATE_FREQ_MS);
 
   useEffect(() => {
-    updateUserLocationToCurrentVenue();
-  }, [updateUserLocationToCurrentVenue]);
-
-  // useEffect(() => {
-  //   if (!userId || !venueName) return;
-  //   updateUserLocationToCurrentVenue();
-
-  // NOTE: A suggestion on how to avoid location cleaning, when two tabs were opened and one of them was closed
+    if (!userId || !venueName) return;
 
-  // document.addEventListener("visibilitychange", () => {
-  //   if (document.visibilityState === "visible") {
-  //     updateUserLocationToCurrentVenue();
-  //   }
-  // });
-  // }, [userId, venueName, updateUserLocationToCurrentVenue]);
+    setNewLocationData({ userId, locationName: venueName });
+  }, [userId, venueName]);
 
   useEffect(() => {
     if (!userId) return;
 
+    const onBeforeUnloadHandler = () => clearLocationData({ userId });
+
     // NOTE: Clear user location on page close
-    window.addEventListener("beforeunload", () =>
-      updateLocationData(userId, {})
-    );
+    window.addEventListener("beforeunload", onBeforeUnloadHandler);
+
+    return () =>
+      window.removeEventListener("beforeunload", onBeforeUnloadHandler);
   }, [userId]);
 
   useEffect(() => {
     if (
-      profile?.enteredVenueIds &&
-      venueId &&
-      profile?.enteredVenueIds.includes(venueId)
-    )
+      !venueId ||
+      !user ||
+      !profile ||
+      profile?.enteredVenueIds?.includes(venueId)
+    ) {
       return;
-    if (!venueId || !user || !profile) return;
+    }
 
     updateProfileEnteredVenueIds(profile?.enteredVenueIds, user?.uid, venueId);
```

Comment:
It seems we already have `userId` defined in this file, so can we narrow the scope of the checks/etc here to use that rather than the full user object?

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1129/commits/53f57c40e24983470b7552d4ebed6f1715ff74ba